source/mod_dist.f90

```diff
@@ -98,159 +265,726 @@ subroutine dist_parseInputLine(inLine, iLine, iErr)
+    call dist_appendFormat(dist_fmtJX,       one,  1)
+    call dist_appendFormat(dist_fmtPhiX,     one,  2)
+    call dist_appendFormat(dist_fmtJY,       one,  3)
+    call dist_appendFormat(dist_fmtPhiY,     one,  4)
+    call dist_appendFormat(dist_fmtJZ,       one,  5)
+    call dist_appendFormat(dist_fmtPhiZ,     one,  6)
+
+  case("IONS") ! The ion columns
+    call dist_appendFormat(dist_fmtMASS,     c1e3, 0)
+    call dist_appendFormat(dist_fmtCHARGE,   one,  0)
+    call dist_appendFormat(dist_fmtIonA,     one,  0)
+    call dist_appendFormat(dist_fmtIonZ,     one,  0)
+    call dist_appendFormat(dist_fmtPDGID,    one,  0)
+
+  case("OLD_DIST") ! The old DIST block file format
+    call dist_appendFormat(dist_fmtPartID,   one,  0)
+    call dist_appendFormat(dist_fmtParentID, one,  0)
+    call dist_appendFormat(dist_fmtNONE,     one,  0)
+    call dist_appendFormat(dist_fmtX,        c1e3, 1)
+    call dist_appendFormat(dist_fmtY,        c1e3, 3)
+    call dist_appendFormat(dist_fmtNONE,     one,  0)
+    call dist_appendFormat(dist_fmtXP,       c1e3, 2)
+    call dist_appendFormat(dist_fmtYP,       c1e3, 4)
+    call dist_appendFormat(dist_fmtNONE,     one,  0)
+    call dist_appendFormat(dist_fmtIonA,     one,  0)
+    call dist_appendFormat(dist_fmtIonZ,     one,  0)
+    call dist_appendFormat(dist_fmtMASS,     c1e3, 0)
+    call dist_appendFormat(dist_fmtP,        c1e3, 6)
+    call dist_appendFormat(dist_fmtDT,       one,  5)
+
+  case default
+    write(lerr,"(a)") "DIST> ERROR Unknown column format '"//trim(fmtName)//"'"
+    fErr = .true.
+
+  end select
+
+end subroutine dist_setColumnFormat
+
+! ================================================================================================ !
+!  Parse File Column Units
+!  V.K. Berglyd Olsen, BE-ABP-HSS
+!  Created: 2019-07-05
+!  Updated: 2019-07-10
+! ================================================================================================ !
+subroutine dist_unitScale(fmtName, fmtUnit, unitType, unitScale, uErr)
+
+  use crcoall
+  use string_tools
+  use numerical_constants
+
+  character(len=*), intent(in)    :: fmtName
+  character(len=*), intent(in)    :: fmtUnit
+  integer,          intent(in)    :: unitType
+  real(kind=fPrec), intent(out)   :: unitScale
+  logical,          intent(inout) :: uErr
+
+  unitScale = zero
+
+  if(unitType == 1) then ! Positions
+    select case(chr_toLower(fmtUnit))
+    case("[m]")
+      unitScale = c1e3
+    case("[mm]")
+      unitScale = one
+    case("[1]")
+      unitScale = one
+    case default
+      goto 100
+    end select
+  elseif(unitType == 2) then ! Angle
+    select case(chr_toLower(fmtUnit))
+    case("[rad]")
+      unitScale = c1e3
+    case("[mrad]")
+      unitScale = one
+    case("[1]")
+      unitScale = one
+    case default
+      goto 100
+    end select
+  elseif(unitType == 3) then ! Energy/Momentum/Mass
+    select case(chr_toLower(fmtUnit))
+    case("[ev]","[ev/c]","[ev/c^2]")
```

Comment:
shall we add also `[ev/c2]` and `[ev/c**2]`?

Reply:
Sure, can do.